packages/block-editor/src/components/block-inspector/index.js

```diff
@@ -129,6 +129,11 @@ const BlockInspectorSingleBlock = ( {
 				</div>
 			) }
 			<InspectorControls.Slot bubblesVirtually={ bubblesVirtually } />
+			<InspectorControls.Slot
+				__experimentalGroup="dimensions"
+				bubblesVirtually={ false }
```

Comment:
Why do we use `bubblesVirtuall = { false }`. Ideally we always use true as we were thinking of deprecating the other approach ultimately.

Reply:
I found when `bubblesVirtually` was `true` it interfered with the registration of panel items via the `ToolsPanel`'s context.

Reply:
It's a know limitation that you need to propagate the context when using slots that bubble virtually. An example when the context is passed down through `fillProps` is for the block controls:

https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/block-controls/slot.js

@diegohaz might remember better the implementation details. I'm not sure how much it differs from the use case you have, but we should try to generalize the solution if possible.

Reply:
Thanks @gziolo, the example you gave has helped get me most of the way to fixing this up.

The only real difference I see in this use case is that the `ToolsPanelContext` isn't being created until the `ToolsPanel` is [when it wraps the Slot](https://github.com/WordPress/gutenberg/pull/34157/files#diff-d54bdb7093990ff38b7f03adf368284bb43cb765ee9100b938c7b5db54e6ba8bR33-R35). I believe this means we need to wrap the Slot again so that component can grab the context and pass it through the `fillProps`.

I tried a HoC approach here but couldn't get that working. Instead, I have a [regular component handling this](https://github.com/WordPress/gutenberg/commit/1659b0ef54e35dadcea995dd6df053a21f96af3a). It almost "works" except that it introduces a rogue `<div>` wrapping all the fills. This breaks the layout of the `ToolsPanel`.  I haven't been able to put my finger on where I've gone wrong here. 

Any help you can offer here would be greatly appreciated 🙏  

Also, would this be something I could continue to iterate on in a separate PR?

Reply:
If everything else is working then you can tackle it separately. It's Important to find a good solution here because the same approach is going to be replicated in other places 😄

Reply:
> The only real difference I see in this use case is that the `ToolsPanelContext` isn't being created until the `ToolsPanel` is [when it wraps the Slot](https://github.com/WordPress/gutenberg/pull/34157/files#diff-d54bdb7093990ff38b7f03adf368284bb43cb765ee9100b938c7b5db54e6ba8bR33-R35). I believe this means we need to wrap the Slot again so that component can grab the context and pass it through the `fillProps`.

How is the component tree structured? Won't the Slot component be rendered inside ToolsPanel, in which case it would have access to the parent context?

Reply:
> If everything else is working then you can tackle it separately. It's Important to find a good solution here because the same approach is going to be replicated in other places 😄

This PR's current state is working well according to the most recent reviews and testing. Myself, @glendaviesnz and @andrewserong have all put it through its paces.

If you are comfortable with this being merged, including the `bubblesVirtually={ false }`, I'll definitely continue iterating on this separately until that good solution is found.

Reply:
This is a bit out of my normal wheelhouse so please excuse my lack of understanding around this.

> How is the component tree structured? Won't the Slot component be rendered inside ToolsPanel, in which case it would have access to the parent context?

That was what I was originally expecting but it only worked when preventing virtual bubbling.

The ToolsPanel creates the context and wraps its children in the context provider. The slot is rendered within that. The panel items are added via that slot. Those panel items need to access the ToolsPanelContext to register themselves with the panel. 

My current understanding is that the slot will have access to the context but the context has to be passed through `fillProps` so that we can re-create the context provider to wrap the fills thereby allowing the panel items access to the context again.

I hope that makes sense!